app/controllers/ticket_purchases_controller.rb

```diff
@@ -30,6 +31,18 @@ def index
 
   private
 
+  def count_purchased_registration_tickets
```

Comment:
Rather than putting this in the controller I think it belongs to the `TicketPurchase.purchase` method, the TicketPurchase model should be the responsible of saying if a purchase is valid or not despite the reason... so, in the controller it will be just one condition more to know what happened, but all the checks should live in the model.